test/clj/rems/api/test_applications.clj

```diff
@@ -432,6 +434,38 @@
     (is (= "application.state/closed"
            (:application/state (get-application application-id user-id))))))
 
+(deftest test-application-submit
+  (let [user-id "alice"
+        form-id (test-data/create-form! {})
+        cat-id (test-data/create-catalogue-item! {:form-id form-id})
+        app-id (test-data/create-application! {:catalogue-item-ids [cat-id] :actor user-id})]
+    (testing "submit with disabled catalogue item fails"
+      (is (:success (catalogue/update-catalogue-item! {:id cat-id
+                                                       :enabled false
+                                                       :archived false})))
+      (rems.db.applications/reset-application-cache!)
```

Comment:
For this test you could stop rems.db.applications/application-cache, in which case the reads would happen without a cache.

Reply:
I think having an explicit cache reset makes the problem more visible so I'll leave it.